dashboard/src/components/branding/che-dashboard-configuration.service.ts

```diff
@@ -41,10 +53,12 @@ export class CheDashboardConfigurationService {
   }
 
   allowRoutes(menuItem: che.ConfigurableMenuItem): ng.IPromise<void> {
-    return this.cheBranding.ready.then(() => {
-      if (this.allowedMenuItem(menuItem) === false) {
-        return this.$q.reject();
-      }
+    return this.$q.resolve().then(() => {
```

Comment:
This will not work as expected. Please consider using and returning deferred object instead.

Reply:
Good catch. Thanks